### ISSUE TITLE
[Feature Request]: Improve Select accessibility

### ISSUE THREAD

**Issue by AykutSarac** [Trendyol/baklava#686]
### Describe the problem or use case

Noticed while using select component for city selection. I wanted to focus directly to the city starting with X, so I pressed to X key however it didn't focus/scroll to it.

### Proposed solution

Should focus to the select option starting with the pressed key

### Additional technical context

Additionally it might listen for other keys while select visible and match the ongoing characters like XY...

**Comment by AykutSarac** [comment]
This should be discussed though whether this should be implemented along with Search functionality or separately, as I was thinking a key press may trigger focus into the search input.

**Comment by muratcorlu** [comment]
This can be considered for the

> This should be discussed though whether this should be implemented along with Search functionality or separately, as I was thinking a key press may trigger focus into the search input.

Good point. I think jumping to the option by writing a name still can be a good improvement when the search is not enabled. But indeed, this can be considered with #265.

**Comment by gokcecicek** [comment]
## Select Accessibility (ADR)
This improvement aims to enable users to access `bl-select-option` elements more easily using the keyboard.

### Rules
- This feature does not work with `search-bar` attribute.
- When a letter key is pressed on the keyboard, if there is an option starting with that letter among the `bl-select-option` elements, that option will be focused automatically.
- When letter keys are pressed consecutively on the keyboard, the first option starting with those letters will be focus.
- If the target option is disabled, it will not be focus when attempting to access it via keyboard.

### Attributes
| Attribute | Description | Default Value | Type |
|--------|--------|--------|--------|
| key-press-threshold | Expected time interval between consecutive key presses on the keyboard | 500 | Number |

**Comment by gokcecicek** [comment]
- I don't have a specific case in mind. I added the `key-press-threshold` attribute to provide more flexibility. I was undecided about this matter, embedding it as 500 directly into the code seems reasonable. We can add it as a prop later if needed
- There is no char limit for consecutive submissions
- You're right! That should provide clarity for anyone reviewing the ADR. I added to ADR that it works case-insensitive.
